psst-gui/src/webapi/client.rs

```diff
@@ -70,29 +65,36 @@ impl WebApi {
         Ok(token.token)
     }
 
-    fn request(&self, method: &str, path: impl Display) -> Result<Request, Error> {
+    fn build_request(&self, 
+        method: &str, 
+        base_url: &str,
+        path: impl Display,
+    ) -> Result<Request, Error> {
         let token = self.access_token()?;
         let request = self
             .agent
-            .request(method, &format!("https://api.spotify.com/{}", path))
+            .request(method, &format!("https://{}/{}", base_url, path))
             .set("Authorization", &format!("Bearer {}", &token));
         Ok(request)
     }
 
-    fn get(&self, path: impl Display) -> Result<Request, Error> {
-        self.request("GET", path)
+    fn request(&self, method: &str, base_url: &str, path: impl Display) -> Result<Request, Error> {
+        self.build_request(method, base_url, path)
     }
 
-    fn put(&self, path: impl Display) -> Result<Request, Error> {
-        self.request("PUT", path)
+    fn get(&self, path: impl Display, base_url: Option<&str>) -> Result<Request, Error> {
+        self.request("GET", base_url.unwrap_or("api.spotify.com"), path)    }
+
+    fn put(&self, path: impl Display, base_url: Option<&str>) -> Result<Request, Error> {
+        self.request("GET", base_url.unwrap_or("api.spotify.com"), path)
     }
 
-    fn post(&self, path: impl Display) -> Result<Request, Error> {
-        self.request("POST", path)
+    fn post(&self, path: impl Display, base_url: Option<&str>) -> Result<Request, Error> {
+        self.request("GET", base_url.unwrap_or("api.spotify.com"), path)
     }
 
-    fn delete(&self, path: impl Display) -> Result<Request, Error> {
-        self.request("DELETE", path)
+    fn delete(&self, path: impl Display, base_url: Option<&str>) -> Result<Request, Error> {
+        self.request("GET", base_url.unwrap_or("api.spotify.com"), path)
```

Comment:
Why were these changes made?

Reply:
These are so I can parse a different base URL which is needed for the custom API request. There's probably a better way of doing this though